tinyexpr.h

```diff
@@ -53,9 +56,9 @@ enum {
 
 typedef struct te_variable {
     const char *name;
-    const void *address;
+    union {const double *address; te_fun0 fun0; te_fun1 fun1; te_fun2 fun2;};
```

Comment:
If you are splitting the address out into different specific function types, why not include all the way up to 7 arguments? And if not, why not just leave it as is with a single void* function and cast for all of them? Currently, I don't think the tinyexpr code uses any functions with more than 2 arguments, but a simple and relatively common example would be a linear interpolation (lerp) function, or an if function like excel's `if(condition,then,else)`.